src/lib.rs

```diff
@@ -85,6 +240,31 @@ impl<T: PartialOrd + Unsigned + CheckedMul> DoubleFactorial<T> for T {
     }
 }
 
+impl<T: Float + FloatConst> FloatDoubleFactorial<T> for T {
```

Comment:
I'm not sure if the (double) factorial for floats makes sense. What are you basing this implementation on?

Reply:
I expanded the docstring, which should hopefully explain this. The regular factorial can be defined using the gamma function for non-integer arguments, but I haven't implemented that. There is also at least double factorial for complex arguments, but that isn't implemented either.